src/components/inputs/Switch/Switch.tsx

```diff
@@ -34,14 +33,14 @@ const Switch = (props: IProps) => {
 		style
 	} = props;
 
-	const [isChecked, setIsChecked] = React.useState(checked)
+	const [checked, setChecked] = React.useState(props.checked)
 
-	React.useEffect(() => setIsChecked(checked), [checked])
+	React.useEffect(() => setChecked(props.checked), [props.checked])
 
 	const handleChange = () => {
-		setIsChecked(prev => !prev);
+		setChecked(!checked);
```

Comment:
Could we keep this as `prev => !prev`? I think it's best practice to use the callback syntax when you need to set a state that depends on the previous state value, since these are asynchronous calls that don't happen immediately

Reply:
You actually could probably do something like: 
```
setIsChecked((prev) => {
    if (onChange) onChange(name, !prev);
    return !prev;
})
```
to guarantee the onChange handler gets the right value

Reply:
I dig it